api/src/main/java/org/apache/iceberg/transforms/Truncate.java

```diff
@@ -243,17 +243,35 @@ public boolean canTransform(Type type) {
 
     @Override
     public UnboundPredicate<CharSequence> project(String name,
-                                                  BoundPredicate<CharSequence> pred) {
-      if (pred.op() == NOT_NULL || pred.op() == IS_NULL) {
-        return Expressions.predicate(pred.op(), name);
+                                                  BoundPredicate<CharSequence> predicate) {
+      switch (predicate.op()) {
+        case NOT_NULL:
+        case IS_NULL:
+          return Expressions.predicate(predicate.op(), name);
+        case STARTS_WITH:
+          if (predicate.literal().value().length() <= width()) {
+            return Expressions.predicate(predicate.op(), name, predicate.literal().value());
+          } else {
+            return ProjectionUtil.truncateArray(name, predicate, this);
+          }
+        default:
+          return ProjectionUtil.truncateArray(name, predicate, this);
       }
-      return ProjectionUtil.truncateArray(name, pred, this);
     }
 
     @Override
     public UnboundPredicate<CharSequence> projectStrict(String name,
                                                         BoundPredicate<CharSequence> predicate) {
-      return null;
+      switch (predicate.op()) {
+        case STARTS_WITH:
+          if (predicate.literal().value().length() <= width()) {
```

Comment:
For the case where the length of the literal equals width, is there value in converting to an equality predicate?

Reply:
I have gone ahead, and converted it into an equality predicate.